### ISSUE TITLE
Custom Auth Flow with passwordless login

### ISSUE THREAD

**Issue by dagovalsusa** [aws-amplify/amplify-flutter#2393]
Docs: https://docs.amplify.aws/lib/auth/signin_with_custom_flow/q/platform/flutter/

_Originally posted by @Jordan-Nelson in https://github.com/aws-amplify/amplify-flutter/issues/410#issuecomment-1137515656_
      

Hello everyone,
I've tried to implement Custom Auth Flow with passwordless login using v0.6.10 of amplify_flutter, but when I call Amplify.Auth.signIn('username') the response is "Cannot issue tokens when ChallengeName is SRP_A".
My Cognito is set right because we have used it in production for 2 years, and without amplify_flutter lib, it works.
Further with amplify_flutter v1.0.0-next1 the method Amplify.Auth.signIn('username') works perfectly.
So, I think that in version 0.6.x there is a bug for the CUSTOM_AUTH flow.

Here my amplifyconfiguration, my piece of code is equal to the tutorial example.

const amplifyconfig = ''' { "UserAgent": "aws-amplify-cli/2.0", "Version": "1.0", "auth": { "plugins": { "awsCognitoAuthPlugin": { "IdentityManager": { "Default": {} }, "CognitoUserPool": { "Default": { "PoolId": "us-east-1_xxxxxxx", "AppClientId": "xxxxxxxxxxxxxxxx", "Region": "us-east-1" } }, "Auth": { "Default": { "authenticationFlowType": "CUSTOM_AUTH", "OAuth": { "WebDomain": "xxxxxxx", "AppClientId": "xxxxxxxxxxxxxxxxx", "SignInRedirectURI": "xxx://", "SignOutRedirectURI": "xxx://", "Scopes": [ "phone", "email", "openid", "profile", "aws.cognito.signin.user.admin" ] } } } } } } }''';

**Comment by dnys1** [comment]
Hi @dagovalsusa, sorry you are facing this issue. Can you try the following in v0.6.10?

```dart
await Amplify.Auth.signIn(
  username: '<username>',
  options: CognitoSignInOptions(authFlowType: AuthenticationFlowType.customAuth),
);
```

**Comment by Jordan-Nelson** [comment]
This issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
